src/parser/ast.rs

```diff
@@ -44,11 +67,85 @@ pub struct BinModifier {
     /// If they are not this field is None.
     pub card: VectorMatchCardinality,
     /// on/ignoring on labels
-    pub matching: VectorMatchModifier,
+    pub matching: Option<VectorMatchModifier>,
     /// If a comparison operator, return 0/1 rather than filtering.
     pub return_bool: bool,
 }
 
+impl Default for BinModifier {
+    fn default() -> Self {
+        Self {
+            card: VectorMatchCardinality::OneToOne,
+            matching: None,
+            return_bool: false,
+        }
+    }
+}
+
+impl BinModifier {
+    pub fn default_modifier() -> Self {
+        Default::default()
+    }
+    pub fn card(mut self, card: VectorMatchCardinality) -> Self {
+        self.card = card;
+        self
+    }
+
+    pub fn matching(mut self, matching: Option<VectorMatchModifier>) -> Self {
+        self.matching = matching;
+        self
+    }
```

Comment:
`card()` and `matching()` sound like getters.
```suggestion
    pub fn with_card(mut self, card: VectorMatchCardinality) -> Self {
        self.card = card;
        self
    }

    pub fn with_matching(mut self, matching: Option<VectorMatchModifier>) -> Self {
        self.matching = matching;
        self
    }
```